lib/logger/lib/logger.ex

```diff
@@ -19,6 +19,27 @@ defmodule Logger do
     * Wraps OTP's `error_logger` to prevent it from
       overflowing.
 
+  A common use case is tracking when an important function is called.
+
+      def solve(problem) do
+        Logger.info fn ->
+          "solve/1 called with problem #{inspect problem}"
+        end
+        solve(problem, [])
+      end
+
+  The `Logger.info/2` macro emits the provided message at the `:info`
+  level. There are additional macros for other levels. Refer to the `:level` and
+  `:compile_time_purge_level` configuration options for more details on how
+  levels work.
+
+  Notice the argument passed to `Logger.info` in the above example is a zero
+  argument function. Although the `Logger` macros accept strings which aren't
+  wrapped in functions, it's not recommended in this case. Calling `"... #{inspect
+  problem}"` without wrapping it in a zero argument function will hurt
+  performance, even when then `:info` level is blocked by the configured runtime
+  logging level.
```

Comment:
What about this rephrasing:

> Although the `Logger` macros accept messages as strings as well as functions, it's recommended to use functions whenever the message is expensive to compute. In the example above, the message will be evaluated (and thus the interpolation inside it) whatever the level is, even if the message will not be actually logged at runtime; the only way of avoiding evaluation of such message is purging the log call at compile-time through the `:compile_time_purge_level` option (see below), or using a function that is evaluated to generate the message only if the message needs to be logged according to the runtime level.

I am not sure this is an improvement though, wdyt?